### PR TITLE
Webinterface Layout

### PR DESCRIPTION
**Pull request by JonahKr**
This is my new Webinterface proposal.

I will try to improve and clean it in the next few weeks.
However i think it is ready to be used.

The changes in Javascript are only related to visibility of objects.

### CODE REVIEW

File: core/interface/static/css/projectalice.css [hunk 1]
```diff
@@ -1,714 +1,246 @@
...
-	color: #d1d1d1;
-	display: flex;
+:root {
+    --primary-color: #eeeeee;
+    --secondary-color: #101014;
+    --log-bg-color: #101014;
+	--bg-color: #fbfbff;
+	--ui-bg-color: #1a1a29;
+	--transit-color: #6f788f;
+	--inv-bg-color:#343647;
```
Comment (Psychokiller1888) [inline core/interface/static/css/projectalice.css]: Codacy Issue found: [Expected RBRACE at line 45, col 2.](https://app.codacy.com/app/ProjectAlice/ProjectAlice/pullRequest?prid=4618587)

**Comment by Psychokiller1888** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 11
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/ProjectAlice/ProjectAlice/pullRequest?prid=4618587&bid=15488171)

File: core/interface/static/css/projectalice.css [hunk 2]
```diff
@@ -1,714 +1,246 @@
...
-	z-index: 1;
-	top: 54px;
-	left: 2px;
-	bottom: 2px;
-	background-color: #1e1e1e;
-	color: #d1d1d1;
-	size: 1em;
+[data-theme="dark"] {
+    --primary-color: #eeeeee;
+    --secondary-color: #eeeeee;
```
Comment (Psychokiller1888) [inline core/interface/static/css/projectalice.css]: Codacy Issue found: [Expected RBRACE at line 50, col 5.](https://app.codacy.com/app/ProjectAlice/ProjectAlice/pullRequest?prid=4618587)

File: core/interface/static/css/projectalice.css [hunk 3]
```diff
@@ -1,714 +1,246 @@
...
-	height: 50px;
-	background-color: #1e1e1e;
-	margin: 2px;
-	color: #d1d1d1;
-	display: flex;
+:root {
+    --primary-color: #eeeeee;
+    --secondary-color: #101014;
+    --log-bg-color: #101014;
+	--bg-color: #fbfbff;
```
Comment (Psychokiller1888) [inline core/interface/static/css/projectalice.css]: Codacy Issue found: [Expected RBRACE at line 42, col 2.](https://app.codacy.com/app/ProjectAlice/ProjectAlice/pullRequest?prid=4618587)

File: core/interface/static/css/projectalice.css [hunk 4]
```diff
@@ -1,714 +1,246 @@
...
-header {
-	font-family: 'marskeregular', Arial, sans-serif;
-	height: 50px;
-	background-color: #1e1e1e;
-	margin: 2px;
-	color: #d1d1d1;
-	display: flex;
+:root {
+    --primary-color: #eeeeee;
+    --secondary-color: #101014;
```
Comment (Psychokiller1888) [inline core/interface/static/css/projectalice.css]: Codacy Issue found: [Expected RBRACE at line 40, col 5.](https://app.codacy.com/app/ProjectAlice/ProjectAlice/pullRequest?prid=4618587)

File: core/interface/static/css/projectalice.css [hunk 5]
```diff
@@ -1,714 +1,246 @@
...
-	background-color: #1e1e1e;
-	color: #d1d1d1;
-	size: 1em;
+[data-theme="dark"] {
+    --primary-color: #eeeeee;
+    --secondary-color: #eeeeee;
+    --log-bg-color: #101014;
+    --bg-color: #343647;
+	--transit-color: #6f788f;
+	--inv-bg-color:#fbfbff;
```
Comment (Psychokiller1888) [inline core/interface/static/css/projectalice.css]: Codacy Issue found: [Expected RBRACE at line 54, col 2.](https://app.codacy.com/app/ProjectAlice/ProjectAlice/pullRequest?prid=4618587)

File: core/interface/static/css/projectalice.css [hunk 6]
```diff
@@ -1,714 +1,246 @@
...
-	font-size: 2em;
-}
-
-.moduleStoreModuleDownloadButton {
-	display: flex;
-}
-
-#closeModuleStore {
-	display: none;
+	min-height: 100% !important;
```
Comment (Psychokiller1888) [inline core/interface/static/css/projectalice.css]: Codacy Issue found: [Use of !important](https://app.codacy.com/app/ProjectAlice/ProjectAlice/pullRequest?prid=4618587)

File: core/interface/static/css/projectalice.css [hunk 7]
```diff
@@ -1,714 +1,246 @@
...
-	font-family: 'marskeregular', Arial, sans-serif;
-	height: 50px;
-	background-color: #1e1e1e;
-	margin: 2px;
-	color: #d1d1d1;
-	display: flex;
+:root {
+    --primary-color: #eeeeee;
+    --secondary-color: #101014;
+    --log-bg-color: #101014;
```
Comment (Psychokiller1888) [inline core/interface/static/css/projectalice.css]: Codacy Issue found: [Expected RBRACE at line 41, col 5.](https://app.codacy.com/app/ProjectAlice/ProjectAlice/pullRequest?prid=4618587)

File: core/interface/static/css/projectalice.css [hunk 8]
```diff
@@ -1,714 +1,246 @@
...
-	left: 2px;
-	bottom: 2px;
-	background-color: #1e1e1e;
-	color: #d1d1d1;
-	size: 1em;
+[data-theme="dark"] {
+    --primary-color: #eeeeee;
+    --secondary-color: #eeeeee;
+    --log-bg-color: #101014;
+    --bg-color: #343647;
```
Comment (Psychokiller1888) [inline core/interface/static/css/projectalice.css]: Codacy Issue found: [Expected RBRACE at line 52, col 5.](https://app.codacy.com/app/ProjectAlice/ProjectAlice/pullRequest?prid=4618587)

File: core/interface/static/css/projectalice.css [hunk 9]
```diff
@@ -1,714 +1,246 @@
...
-	margin: 2px;
-	color: #d1d1d1;
-	display: flex;
+:root {
+    --primary-color: #eeeeee;
+    --secondary-color: #101014;
+    --log-bg-color: #101014;
+	--bg-color: #fbfbff;
+	--ui-bg-color: #1a1a29;
+	--transit-color: #6f788f;
```
Comment (Psychokiller1888) [inline core/interface/static/css/projectalice.css]: Codacy Issue found: [Expected RBRACE at line 44, col 2.](https://app.codacy.com/app/ProjectAlice/ProjectAlice/pullRequest?prid=4618587)

File: core/interface/static/css/projectalice.css [hunk 10]
```diff
@@ -1,714 +1,246 @@
...
 
-header {
-	font-family: 'marskeregular', Arial, sans-serif;
-	height: 50px;
-	background-color: #1e1e1e;
-	margin: 2px;
-	color: #d1d1d1;
-	display: flex;
+:root {
+    --primary-color: #eeeeee;
```
Comment (Psychokiller1888) [inline core/interface/static/css/projectalice.css]: Codacy Issue found: [Expected RBRACE at line 39, col 5.](https://app.codacy.com/app/ProjectAlice/ProjectAlice/pullRequest?prid=4618587)

File: core/interface/static/css/projectalice.css [hunk 11]
```diff
@@ -1,714 +1,246 @@
...
+    --secondary-color: #101014;
+    --log-bg-color: #101014;
+	--bg-color: #fbfbff;
+	--ui-bg-color: #1a1a29;
```
Comment (Psychokiller1888) [inline core/interface/static/css/projectalice.css]: Codacy Issue found: [Expected RBRACE at line 43, col 2.](https://app.codacy.com/app/ProjectAlice/ProjectAlice/pullRequest?prid=4618587)

File: core/interface/static/css/projectalice.css (hunk 1 above)
Comment (JonahKr) [reply core/interface/static/css/projectalice.css]: This issue is no issue... its just a use of variables.

**Comment by Psychokiller1888** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 10
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/ProjectAlice/ProjectAlice/pullRequest?prid=4618587&bid=15488171)

File: core/interface/static/css/projectalice.css [hunk 12]
```diff
@@ -1,714 +1,246 @@
...
-	background-color: #1e1e1e;
-	margin: 2px;
-	color: #d1d1d1;
-	display: flex;
+:root {
+	--primary-color: #eeeeee;
+	--secondary-color: #101014;
+	--log-bg-color: #101014;
+	--bg-color: #fbfbff;
+	--ui-bg-color: #1a1a29;
```
Comment (Psychokiller1888) [inline core/interface/static/css/projectalice.css]: Codacy Issue found: [Expected RBRACE at line 43, col 2.](https://app.codacy.com/app/ProjectAlice/ProjectAlice/pullRequest?prid=4618587)

File: core/interface/static/css/projectalice.css [hunk 13]
```diff
@@ -1,714 +1,246 @@
...
-	left: 2px;
-	bottom: 2px;
-	background-color: #1e1e1e;
-	color: #d1d1d1;
-	size: 1em;
+[data-theme="dark"] {
+	--primary-color: #eeeeee;
+	--secondary-color: #eeeeee;
+	--log-bg-color: #101014;
+	--bg-color: #343647;
```
Comment (Psychokiller1888) [inline core/interface/static/css/projectalice.css]: Codacy Issue found: [Expected RBRACE at line 52, col 5.](https://app.codacy.com/app/ProjectAlice/ProjectAlice/pullRequest?prid=4618587)

File: core/interface/static/css/projectalice.css [hunk 14]
```diff
@@ -1,714 +1,246 @@
...
+	--transit-color: #6f788f;
+	--inv-bg-color: #fbfbff;
```
Comment (Psychokiller1888) [inline core/interface/static/css/projectalice.css]: Codacy Issue found: [Expected RBRACE at line 54, col 2.](https://app.codacy.com/app/ProjectAlice/ProjectAlice/pullRequest?prid=4618587)

File: core/interface/static/css/projectalice.css [hunk 15]
```diff
@@ -1,714 +1,246 @@
...
-header {
-	font-family: 'marskeregular', Arial, sans-serif;
-	height: 50px;
-	background-color: #1e1e1e;
-	margin: 2px;
-	color: #d1d1d1;
-	display: flex;
+:root {
+	--primary-color: #eeeeee;
+	--secondary-color: #101014;
```
Comment (Psychokiller1888) [inline core/interface/static/css/projectalice.css]: Codacy Issue found: [Expected RBRACE at line 40, col 5.](https://app.codacy.com/app/ProjectAlice/ProjectAlice/pullRequest?prid=4618587)

File: core/interface/static/css/projectalice.css [hunk 16]
```diff
@@ -1,714 +1,246 @@
...
+	--log-bg-color: #101014;
+	--bg-color: #fbfbff;
```
Comment (Psychokiller1888) [inline core/interface/static/css/projectalice.css]: Codacy Issue found: [Expected RBRACE at line 42, col 2.](https://app.codacy.com/app/ProjectAlice/ProjectAlice/pullRequest?prid=4618587)

File: core/interface/static/css/projectalice.css [hunk 17]
```diff
@@ -1,714 +1,246 @@
...
-	font-family: 'marskeregular', Arial, sans-serif;
-	height: 50px;
-	background-color: #1e1e1e;
-	margin: 2px;
-	color: #d1d1d1;
-	display: flex;
+:root {
+	--primary-color: #eeeeee;
+	--secondary-color: #101014;
+	--log-bg-color: #101014;
```
Comment (Psychokiller1888) [inline core/interface/static/css/projectalice.css]: Codacy Issue found: [Expected RBRACE at line 41, col 5.](https://app.codacy.com/app/ProjectAlice/ProjectAlice/pullRequest?prid=4618587)

File: core/interface/static/css/projectalice.css [hunk 18]
```diff
@@ -1,714 +1,246 @@
...
+	--bg-color: #fbfbff;
+	--ui-bg-color: #1a1a29;
+	--transit-color: #6f788f;
```
Comment (Psychokiller1888) [inline core/interface/static/css/projectalice.css]: Codacy Issue found: [Expected RBRACE at line 44, col 2.](https://app.codacy.com/app/ProjectAlice/ProjectAlice/pullRequest?prid=4618587)

File: core/interface/static/css/projectalice.css [hunk 19]
```diff
@@ -1,714 +1,246 @@
...
-	z-index: 1;
-	top: 54px;
-	left: 2px;
-	bottom: 2px;
-	background-color: #1e1e1e;
-	color: #d1d1d1;
-	size: 1em;
+[data-theme="dark"] {
+	--primary-color: #eeeeee;
+	--secondary-color: #eeeeee;
```
Comment (Psychokiller1888) [inline core/interface/static/css/projectalice.css]: Codacy Issue found: [Expected RBRACE at line 50, col 5.](https://app.codacy.com/app/ProjectAlice/ProjectAlice/pullRequest?prid=4618587)

File: core/interface/static/css/projectalice.css [hunk 20]
```diff
@@ -1,714 +1,246 @@
...
-	color: #d1d1d1;
-	display: flex;
+:root {
+	--primary-color: #eeeeee;
+	--secondary-color: #101014;
+	--log-bg-color: #101014;
+	--bg-color: #fbfbff;
+	--ui-bg-color: #1a1a29;
+	--transit-color: #6f788f;
+	--inv-bg-color: #343647;
```
Comment (Psychokiller1888) [inline core/interface/static/css/projectalice.css]: Codacy Issue found: [Expected RBRACE at line 45, col 2.](https://app.codacy.com/app/ProjectAlice/ProjectAlice/pullRequest?prid=4618587)

File: core/interface/static/css/projectalice.css [hunk 21]
```diff
@@ -1,714 +1,246 @@
...
 
-header {
-	font-family: 'marskeregular', Arial, sans-serif;
-	height: 50px;
-	background-color: #1e1e1e;
-	margin: 2px;
-	color: #d1d1d1;
-	display: flex;
+:root {
+	--primary-color: #eeeeee;
```
Comment (Psychokiller1888) [inline core/interface/static/css/projectalice.css]: Codacy Issue found: [Expected RBRACE at line 39, col 5.](https://app.codacy.com/app/ProjectAlice/ProjectAlice/pullRequest?prid=4618587)

**Comment by JonahKr** [comment]
The Errors by Codacy are related to the use of variables... 
it expects brackets there but it works totally fine.

**Comment by JonahKr** [comment]
https://www.w3schools.com/css/css3_variables.asp

**Comment by Psychokiller1888** [comment]
Cool, thank you! Will test this now! The only problem, is that you worked on the main alpha branch, it's a pain to test... Can you branch your changes to a new work branch so we can easily switch between yours and the official?

**Comment by JonahKr** [comment]
i will close this request and create a new branch and will implement some fixes

**Pull request closed without merging**